descqa/DensityVersusExtinction.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import unicode_literals, absolute_import, division
+import os
+import numpy as np
+from scipy.stats import binned_statistic
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+import healpy as hp
+
+__all__ = ['DensityVersusExtinction']
+
+def create_hp_map(ra, dec, nside):
+    """
+    Auxiliary function to generate HEALPix maps from catalogs.
+    It reads the ra and dec in degrees and returns a HEALPix map
+    """
+    pixnums = hp.ang2pix(nside, np.pi/2-np.radians(dec), np.radians(ra))
+    return np.bincount(pixnums, minlength=hp.nside2npix(nside))
+
+class DensityVersusExtinction(BaseValidationTest):
+    """
+    This tests checks that the object density decreases with increasing
+    MW extinction
+    """
+    def __init__(self,**kwargs): # pylint: disable=W0231
+        self.kwargs = kwargs
+        self.test_name = kwargs['test_name']
+        self.validation_path = kwargs['validation_map_filename']
+        self.nside = kwargs['nside'] 
+        self.validation_data = hp.ud_grade(hp.read_map(self.validation_path), nside_out=self.nside)
+        self.xlabel = kwargs['xlabel']
+     
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        
+        catalog_data = catalog_instance.get_quantities(['ra', 'dec', 'extendedness'])
+        galaxies = catalog_data['extendedness']==1 
+        data_map = create_hp_map(catalog_data['ra'][galaxies], catalog_data['dec'][galaxies], self.nside)
```

Comment:
Lines 34-36: (1) add `has_quantities` and skip test if not all needed quantities exist, (2) use `filters` instead of masking the data afterwards.
```python
        if not catalog_instance.has_quantities(['ra', 'dec', 'extendedness']):
            return TestResult(skipped=True, summary='catalog does not have needed quantities')
        catalog_data = catalog_instance.get_quantities(['ra', 'dec', 'extendedness'], filters=['extendedness == 1'])
        data_map = create_hp_map(catalog_data['ra'], catalog_data['dec'], self.nside)
```